kucherov_vasilii/graph_json_printer.cpp

```diff
@@ -39,26 +40,16 @@ class Graph {
   void add_vertex() { vertices_.emplace_back(get_new_vertex_id()); }
 
   void add_edge(VertexId from_vertex_id, VertexId to_vertex_id) {
-    edges_.emplace_back(get_new_edge_id(), from_vertex_id, to_vertex_id);
-  }
-
-  std::vector<EdgeId> get_vertex_edges_ids(const Graph::Vertex& vertex) const {
-    std::vector<EdgeId> edge_ids;
+    auto edge_id = get_new_edge_id();
 
-    for (const auto& edge : edges_) {
-      if (edge.from_vertex_id() == vertex.id() ||
-          edge.to_vertex_id() == vertex.id()) {
-        edge_ids.push_back(edge.id());
-      }
-    }
+    edges_.emplace_back(edge_id, from_vertex_id, to_vertex_id);
 
-    return edge_ids;
+    adjacency_list_[from_vertex_id].push_back(edge_id);
+    adjacency_list_[to_vertex_id].push_back(edge_id);
   }
 
-  std::vector<VertexId> get_edge_vertices_ids(const Graph::Edge& edge) const {
-    std::vector<VertexId> vertex_ids = {edge.from_vertex_id(),
-                                        edge.to_vertex_id()};
-    return vertex_ids;
+  const std::vector<EdgeId>& get_connected_edge_ids(VertexId vertex_id) const {
+    return adjacency_list_.at(vertex_id);
```

Comment:
По вашему вопросу, как вариант, можно сделать так:
```suggestion
    if (!adjacency_list_.has(vertex_id)) {
      static const std::vector<EdgeId> empty_result;
      return empty_result;
    }
    return adjacency_list_.at(vertex_id);
```